src/lime/media/AudioSource.hx

```diff
@@ -36,6 +37,8 @@ class AudioSource
 
 		this.loops = loops;
 
+		pitch = 1;
```

Comment:
I mean, we could do that, but then shouldn't we also add `gain` and `position` arguments? You gotta draw a line at some point, or you start to end up with [nonsense like this](https://github.com/openfl/openfl/blob/develop/src/openfl/text/TextFormat.hx#L255-L257).

My instinct is that `pitch` is a niche feature, and doesn't need to be in the constructor. At least, not until we hear from lots more people using it.

Reply:
> I mean, we could do that, but then shouldn't we also add `gain` and `position` arguments? You gotta draw a line at some point, or you start to end up with [nonsense like this](https://github.com/openfl/openfl/blob/develop/src/openfl/text/TextFormat.hx#L255-L257).
> 
> My instinct is that `pitch` is a niche feature, and doesn't need to be in the constructor. At least, not until we hear from lots more people using it.

Hm, Seems fair to say that

Reply:
Yeah I think sounds will almost always be initialized as their default pitch of 1. 
If there is any other interpretation of how to initialize that variable though that follows lime more closely would be good for little consistency like that.

Reply:
> Yeah I think sounds will almost always be initialized as their default pitch of 1. If there is any other interpretation of how to initialize that variable though that follows lime more closely would be good for little consistency like that.

Maybe we should get rid of `pitch = 1;` thing? with the review comments above included of course

Reply:
Ok, I know why this is crashing. It's because you're calling for the backend but the backend isn't called yet so it's null, and making it crash.
You should put it next to the length variable and it won't crash

Reply:
Wait, but it's already in order. `__backend` is set first, then `length`, then `pitch`.

Are you sure it's an error that `__backend` is null, and not `handle`? Because I noticed an error with the latter...

Reply:
> Wait, but it's already in order. `__backend` is set first, then `length`, then `pitch`.
> 
> Are you sure it's an error that `__backend` is null, and not `handle`? Because I noticed an error with the latter...

No, Forget about it because I'm stupid and I placed the thing on top of the `__backend` cos I'm stupid

Reply:
Ok, so it all works now?

Reply:
Well, The pitch goes great with the `if (handle != null)` command so I guess?

I'll go beyond and check if it's really needed the handle thing. But at the moment, like it's set, It is

Reply:
Yeah, that's expected. `handle` is null until `init()` is called, and `init()` is only called after `pitch = 1`. With the `handle != null` check, it simply ignores changes that happen before `init()`, meaning `pitch = 1` has no effect on `NativeAudioSource`.

`HTML5AudioSource` uses howler.js, which [sets pitch to 1 by default](https://github.com/goldfire/howler.js/blob/f128f9d975c2bc87610eac0eab5239980b3a4dda/dist/howler.js#L589).

Oh, and `FlashAudioSource` ignores pitch.

Conclusion: `pitch = 1` is useless no matter what.

Reply:
Maybe we should just get rid of pitch?
It seems that it's just code that doesn't do anything

Reply:
Or at least place the pitch below the `init` function

Reply:
Yeah, I scrapped it.